src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java

```diff
@@ -162,103 +171,127 @@ public void validateIncompatibleJavadocCheckNames() throws IOException {
-                                + " Please update the todo list in"
-                                + " XpathRegressionTest.INCOMPATIBLE_CHECK_NAMES")
-                        .that(INCOMPATIBLE_CHECK_NAMES.contains(check))
-                        .isFalse();
-                compatibleChecks.add(check);
-            }
-        }
-
         // Ensure that all lists are up-to-date
         final Set<String> allChecks = new HashSet<>(SIMPLE_CHECK_NAMES);
         allChecks.removeAll(INCOMPATIBLE_JAVADOC_CHECK_NAMES);
         allChecks.removeAll(INCOMPATIBLE_CHECK_NAMES);
         allChecks.removeAll(Set.of("Regexp", "RegexpSinglelineJava", "NoCodeInFile"));
         allChecks.removeAll(NO_VIOLATION_MODULES);
-        allChecks.removeAll(compatibleChecks);
+        allChecks.removeAll(compatibleChecks());
         allChecks.removeAll(INTERNAL_MODULES);
 
         assertWithMessage("XpathRegressionTest is missing for [" + String.join(", ", allChecks)
                 + "]. Please add them to src/it/java/org/checkstyle/suppressionxpathfilter")
-                        .that(allChecks)
-                        .isEmpty();
+                .that(allChecks)
+                .isEmpty();
     }
 
-    @Test
-    public void validateInputFiles() throws Exception {
-        try (DirectoryStream<Path> dirs = Files.newDirectoryStream(inputDir)) {
-            for (Path dir : dirs) {
-                // input directory must be named in lower case
-                assertWithMessage(dir + " is not a directory")
-                        .that(Files.isDirectory(dir))
-                        .isTrue();
-                final String dirName = dir.toFile().getName();
-                assertWithMessage("Invalid directory name: " + dirName)
-                        .that(ALLOWED_DIRECTORY_AND_CHECKS)
-                        .containsKey(dirName);
-
-                // input directory must be connected to an existing test
-                final String check = ALLOWED_DIRECTORY_AND_CHECKS.get(dirName);
-                final Path javaPath = javaDir.resolve("XpathRegression" + check + "Test.java");
-                assertWithMessage("Input directory '" + dir
-                            + "' is not connected to Java test case: " + javaPath)
-                        .that(Files.exists(javaPath))
-                        .isTrue();
+    private Set<String> compatibleChecks() throws IOException {
+        try (Stream<Path> files = Files.list(javaDir)) {
+            return files
+                    .filter(Files::isRegularFile)
+                    .map(Path::getFileName)
+                    .map(Path::toString)
+                    .filter(name -> !name.startsWith("Abstract"))
+                    .flatMap(name -> firstGroupMatchOrEmpty(REGRESSION_TEST_NAME.matcher(name)))
+                    .map(XpathRegressionTest::checkForSimpleName)
+                    .map(XpathRegressionTest::checkForIncompatibleName)
```

Comment:
[Single-Responsibility-Prinzip](https://de.wikipedia.org/wiki/Single-Responsibility-Prinzip) (one thing only)

Reply:
please come back to my recommendation on design of methods.

Reply:
inline as lambda and merge `checkForSimpleName` + `checkForIncompatibleName`